include/nccl_ofi_config_bottom.h

```diff
@@ -9,7 +9,11 @@
 #define OFI_LIKELY(x)   __builtin_expect((x), 1)
 #define OFI_UNLIKELY(x) __builtin_expect((x), 0)
 
+#ifdef __cplusplus
+#define NCCL_OFI_EXPORT_SYMBOL __attribute__((visibility("default"))) [[maybe_unused]]
```

Comment:
What was the compiler warning leading to this?  This seems like the wrong annotation for public symbols.

Reply:
Cannot recall, sorry. This is harmless and has no effect other than silencing unused warnings.

I know [GenerateExportHeader](https://cmake.org/cmake/help/latest/module/GenerateExportHeader.html) will create a more complete export macro than what we have, and I suspect we're missing some of that magic in ours. This was enough to get the build to succeed for me.